lib/thesis/stores/ecto_store.ex

```diff
@@ -114,6 +140,20 @@ defmodule Thesis.EctoStore do
     :ok
   end
 
+  defp save_backup(nil, _), do: :error
+  defp save_backup(page_id, page_data) do
+    backups = backups(page_id)
+
+    backup_changeset = Backup.changeset(%Backup{}, %{
+      page_id: page_id,
+      page_revision: new_backup_page_revision(backups),
+      page_data:  LZString.compress(page_data)
+    })
+
+    repo.insert!(backup_changeset)
+    :ok
```

Comment:
Same as my previous comment; We should return the result of `repo.insert!` or check its result and behave accordingly.

Also, `insert!` raises an error on failure, rather than returning an error. I think we should handle that error case.